modules/gateway/rpc.go

```diff
@@ -162,16 +198,24 @@ func (g *Gateway) Broadcast(name string, obj interface{}, peers []modules.Peer)
 	}
 
 	var wg sync.WaitGroup
-	wg.Add(len(peers))
 	for _, p := range peers {
+		wg.Add(1)
 		go func(addr modules.NetAddress) {
+			defer wg.Done()
 			err := g.RPC(addr, name, fn)
 			if err != nil {
+				g.log.Debugf("WARN: broadcasting RPC %q to peer %q failed (attempting again in 10 seconds): %v", name, addr, err)
 				// try one more time before giving up
-				time.Sleep(10 * time.Second)
-				g.RPC(addr, name, fn)
+				select {
+				case <-time.After(10 * time.Second):
```

Comment:
10 seconds is both not long enough and is a magical number. Can we create a const for this value? Also, the gateway I think has enough files that it deserves a `consts.go`, but that should be done in a different PR.